src/gui/map/map_information_dialog.cpp

```diff
@@ -0,0 +1,395 @@
+	
+	if (map_information.undo_steps_num || map_information.redo_steps_num)
+	{
+		QString undo_redo_type = map_information.undo_steps_num ? tr("Undo") : tr("Redo");
+		QString undo_redo_num = QString::fromLatin1("%1").arg(map_information.undo_steps_num ? map_information.undo_steps_num : map_information.redo_steps_num);
+		if (map_information.undo_steps_num && map_information.redo_steps_num)
+		{
+			undo_redo_type.append(QLatin1Char('/')).append(tr("Redo"));
+			undo_redo_num.append(QString::fromLatin1("/%1").arg(map_information.redo_steps_num));
+		}
+		undo_redo_type.append(QChar::Space).append(tr("steps"));
+		undo_redo_num.append(QChar::Space).append(map_information.undo_steps_num + map_information.redo_steps_num > 1 ? tr("steps") : tr("step"));
+		addTreeItem(0, undo_redo_type, undo_redo_num);
+	}
+	
+	addTreeItem(0, tr("Colors"), tr("%n color(s)", nullptr, map_information.colors_num));
+	if (map_information.colors_num)
+	{
+		for (const auto& color : map_information.colors)
+		{
+			addTreeItem(1, color.name);
+			if (color.symbols.size())
+			{
+				for (const auto& symbol : color.symbols)
+				{
+					addTreeItem(2, symbol);
+				}
+			}
+		}
+	}
+	
+	addTreeItem(0, tr("Fonts"), tr("%n font(s)", nullptr, map_information.fonts_num));
+	for (const auto& font_name : map_information.font_names)
+	{
+		addTreeItem(1, (font_name.name == font_name.name_substitute ? font_name.name : font_name.name + tr(" (substituted by ") + font_name.name_substitute + QLatin1Char(')')), 
+		            tr("%n symbol(s)", nullptr, font_name.num));
+	}
+	
+	addTreeItem(0, tr("Objects"), tr("%n object(s)", nullptr, map_information.map_objects_num));
+	for (const auto& map_object : map_information.map_objects)
+	{
+		if (map_object.category.num)
+		{
+			addTreeItem(1, map_object.category.name, tr("%n object(s)", nullptr, map_object.category.num));
+			for (const auto& object : map_object.objects)
+			{
+				addTreeItem(2, object.name, tr("%n object(s)", nullptr, object.num));
+				if (object.colors.size())
+				{
+					for (const auto& color : object.colors)
+					{
+						addTreeItem(3, color);
+					}
+				}
+			}
+		}
+	}
+}
+
+void MapInformationDialog::addTreeItem(const int level, const QString& item, const QString& value)
+{
+	Q_ASSERT(level >= 0 && level <= 3);
+	
+	tree_items.push_back({level, item, value});
+	max_item_length = qMax(max_item_length, level * text_report_indent + item.length());
+}
+
+void MapInformationDialog::setupTreeWidget()
+{
+	for (const auto &tree_item : tree_items)
+	{
+		auto tree_depth = static_cast<int>(tree_item_hierarchy.size());
+		Q_ASSERT(tree_item.level <= tree_depth);
+	
+		for (; tree_depth > tree_item.level; --tree_depth)
+			tree_item_hierarchy.pop_back();
+	
+		QTreeWidgetItem* tree_widget_item;
+		if (tree_depth)
+			tree_widget_item = new QTreeWidgetItem(tree_item_hierarchy.back());
+		else
+			tree_widget_item = new QTreeWidgetItem(map_info_tree);
```

Comment:
My style would be 
`auto* tree_widget_item  = new QTreeWidgetItem(tree_depth ? tree_widget_itemtree_item_hierarchy.back() : map_info_tree);`

And yes, not everyone likes it. :-)

Reply:
Especially the compiler does not like it ;-)
I tried that before but had to learn that the compiler apparently wants to know at compile time which constructor to use.
This variation is accepted:
`auto* tree_widget_item  = tree_depth ? new QTreeWidgetItem(tree_item_hierarchy.back()) : new QTreeWidgetItem(map_info_tree);`

Reply:
Is that equivalent:
```suggestion
		auto* tree_widget_item = new QTreeWidgetItem (tree_depth ? tree_item_hierarchy.back() : map_info_tree.invisibleRootItem());
```